lib/chef/provider/systemd_unit.rb

```diff
@@ -58,20 +58,28 @@ def define_resource_requirements
 
       def systemd_unit_status
         @systemd_unit_status ||= begin
+          status = {}
+
           # Collect all the status information for a unit and return it at once
-          # This may fail if we are managing a template unit (e.g. with '@'), in which case
-          # we just ignore the error because unit status is irrelevant in that case
+          # This will fail if we are managing a template unit (e.g. with '@'), which is why
+          # we capture 'is-enabled' below
+          # This fails because '--system show' returns 'enabled' for *all* templated units
+          # if *any* of the templated units are enabled
+          # e.g. if daemon@env1 is enabled daemon@env2 will also show enabled,
+          # whether or not daemon@env2 is actually enabled
           s = shell_out(*systemctl_cmd, "show", "-p", "UnitFileState", "-p", "ActiveState", new_resource.unit_name, **systemctl_opts)
           # e.g. /bin/systemctl --system show -p UnitFileState -p ActiveState syslog.socket
           # Returns something like:
           # ActiveState=inactive
           # UnitFileState=static
-          status = {}
           s.stdout.each_line do |line|
             k, v = line.strip.split("=")
             status[k] = v
           end
 
+          status["is-enabled"] = shell_out(*systemctl_cmd, "is-enabled", new_resource.unit_name, **systemctl_opts).stdout.strip
+          # Returns "enabled" or "disabled"
```

Comment:
It'll return a bunch of other things, like "static", "generated", "transient", etc.  It's probably better to check the exit code (which I believe was the prior behavior) since that's more binary, though I think the change in `enabled?` technically covers it.

Reply:
@rpunt thoughts on this?

Reply:
OK, so do we want to drop this? @rpunt

Reply:
I guess I'm not particularly tied to the method of resolution, so long as their *is* a resolution.